nautobot/dcim/filters.py

```diff
@@ -547,9 +523,14 @@ class Meta:
 
 
 class DeviceFilterSet(NautobotFilterSet, TenancyFilterSet, LocalContextFilterSet, StatusModelFilterSetMixin):
-    q = django_filters.CharFilter(
-        method="search",
-        label="Search",
+    q = SearchFilter(
+        filter_predicates={
+            "name": "icontains",
+            "serial": "icontains",
```

Comment:
Looking at the code that was removed I'm wondering if these and the following two fields desire to be `iexact` instead. This likely affects tests but food for thought none the less.

Reply:
`iexact` with `lamba v: v.strip()` as preprocessor?

Reply:
I think it's reasonable for these to be `icontains` rather than `iexact`. If there's a common pattern to serial numbers or asset tags, I can absolutely see a substring search being useful to a user.